src/iterable.js

```diff
@@ -58,18 +59,22 @@ function handleItem (resolve, handler, x, i, promise) {
 	} else if (isRejected(p)) {
 		handler.rejectAt(p, i, promise)
 	} else {
-		settleAt(p, handler, i, promise)
+		p._runAction(new Indexed(handler, i, promise))
```

Comment:
:+1:

I had noticed that the object literal gave slightly better performance than using a class.  However, now I realize it was probably due to the `_classCallCheck` hit.  Now that it's gone, this is much nicer.